### PR TITLE
add apiproduct operations sample

### PR DESCRIPTION
**Pull request by DinoChiesa**
This adds the API Product operations sample, including a README, a tutorial file, sample API Proxies, and setup and cleanup scripts.

### CODE REVIEW

File: apiproduct-operations/README.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+the proxy. All of the magic of authorizing the operation for the given
+credential is done _implicitly_ by the VerifyAPIKey policy. There's nothing else
+you need to do in the API Proxy to make that happen.
+
+For the purposes of demonstration, the proxy uses an
+[AssignMessage](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy)
+policy to set a mock success response, which includes the API product name, and
+the operation that was authorized.
+
+There's also a conditional flow in the proxy that uses [OAuthV2/VeriryAccessToken](https://cloud.google.com/apigee/docs/api-platform/reference/policies/oauthv2-policy#verifyaccesstoken) in place of VerifyAPIKey.
```
Comment (ssvaidyanathan) [inline apiproduct-operations/README.md:100]: typo VerifyAccessToken

File: apiproduct-operations/README.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    * unzip
+    * curl
+    * jq
+    * npm
+
+## (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/DinoChiesa/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=apiproduct-operations/docs/cloudshell-tutorial.md)
```
Comment (ssvaidyanathan) [inline apiproduct-operations/README.md:124]: the URL is pointing to your personal repo

File: apiproduct-operations/README.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+## Manual Setup instructions
+
+If you've clicked the blue button above, you can ignore the rest of this README.
+If you choose _not_ to follow the tutorial in Cloud Shell, you can follow these steps on your own:
+
+1. Clone the `apigee-samples` repo, and cd into the `apiproduct-operations` directory
+
+   ```bash
+   git clone https://github.com/GoogleCloudPlatform/apigee-samples.git
+   cd b/apiproduct-operations
```
Comment (ssvaidyanathan) [inline apiproduct-operations/README.md:135]: is the `b/` a typo?

File: apiproduct-operations/img/screenshot-20230828-140220.png
Comment (ssvaidyanathan) [inline apiproduct-operations/img/screenshot-20230828-140220.png:1]: usually the screenshot is added when we have a YT video uploaded. I would suggest we remove this and then add it once we have the YT video uploaded. We can refer to the img pointing to YT itself. You can refer other samples

**Comment by ssvaidyanathan** [comment]
Also pls fix the issues raised by the lint

**Comment by DinoChiesa** [comment]
ok I tried to fix all the linter issues, and the other things you pointed out.

**Comment by DinoChiesa** [comment]
ok, this is again ready for review.

File: apiproduct-operations/README.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+For the purposes of demonstration, the proxy uses an
+[AssignMessage](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy)
+policy to set a mock success response, which includes the API product name, and
+the operation that was authorized.
+
+There's also a conditional flow in the proxy that uses [OAuthV2/VerifyAccessToken](https://cloud.google.com/apigee/docs/api-platform/reference/policies/oauthv2-policy#verifyaccesstoken) in place of VerifyAPIKey.
+The behavior with regard to operations authorization checks is exactly the same.
+
+
+## Screencast
```
Comment (ssvaidyanathan) [inline apiproduct-operations/README.md:104]: I would recommend removing this section and adding it when you have the video ready

**Comment by ssvaidyanathan** [comment]
> ok, this is again ready for review.

Can you also resolve the review comments I posted

File: apiproduct-operations/README.md (hunk 3 above)
Comment (DinoChiesa) [reply apiproduct-operations/README.md]: yes it was a typo

File: apiproduct-operations/README.md (hunk 2 above)
Comment (DinoChiesa) [reply apiproduct-operations/README.md]: fixed.

File: apiproduct-operations/README.md (hunk 1 above)
Comment (DinoChiesa) [reply apiproduct-operations/README.md]: fixed

File: apiproduct-operations/README.md (hunk 4 above)
Comment (DinoChiesa) [reply apiproduct-operations/README.md]: fixed with latest commit

**Comment by ssvaidyanathan** [comment]
LGTM - are the bash scripts tested?

**Pull request merged**